elytron/src/main/java/org/wildfly/extension/elytron/PolicyDefinitions.java

```diff
@@ -250,6 +261,18 @@ protected void recordCapabilitiesAndRequirements(OperationContext context, Model
             @Override
             public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
                 OperationStepHandler write = new ReloadRequiredWriteAttributeHandler(attributes) {
+
+                    @Override
+                    protected boolean applyUpdateToRuntime(OperationContext context, ModelNode operation, String attributeName, ModelNode resolvedValue, ModelNode currentValue, HandbackHolder<Void> voidHandback) throws OperationFailedException {
+                        // the configuration factory will only be applied on restart
+                        for (ModelNode modelNode : resolvedValue.asList()) {
+                            if (modelNode.asProperty().getName().equals(CONFIGURATION_FACTORY)) {
```

Comment:
I believe this handler is used for DEFAULT_POLICY, JaccPolicyDefinition.POLICY, CustomPolicyDefinition.POLICY. The resolvedValue.asList will break fro DEFAULT_POLICY, and the CONFIGURATION_FACTORY field is not relevant to CustomPolicyDefinition.POLICY.

Can this just check the value of attributeName?

Reply:
Further to this, I was going to comment on how reverting this restartRequired needs to be handled as well. But I think this entire method override can be removed by just adding a call to setRestartJvm() to the builder that builds JaccPolicyDefinition.POLICY

Reply:
I missed that it's applied to other policies, sorry.
As for using setRestartJVM() - that would apply the restartRequired to changes to other attributes as well. I was trying to restrict this to only CONFIGURATION_FACTORY, but if that's not a problem, I'll make this change.